crosshair/libimpl/builtinslib.py

```diff
@@ -1967,6 +1990,24 @@ def split(self, sep: Optional[str] = None, maxsplit: int = -1):
         ret.extend(self[first_occurance + 1 :].split(sep=sep, maxsplit=new_maxsplit))
         return ret
 
+    def rsplit(self, sep: Optional[str] = None, maxsplit: int = -2):
+        if sep is None:
+            return self.__str__().split(sep=sep, maxsplit=maxsplit)
+        smt_sep = force_to_smt_sort(sep, SmtStr)
+        if not isinstance(maxsplit, Integral):
+            raise TypeError
+        if maxsplit == 0:
+            return [self]
+        last_occurence = self.rfind(sep)
+        if last_occurence == -1:
+            return [self]
+        new_maxsplit = -1 if maxsplit == -1 else maxsplit - 1
```

Comment:
```suggestion
        new_maxsplit = -1 if maxsplit < 0 else maxsplit - 1
```

I think this is wrong in my version of `split()` too - Python seems to allow the user to pass in `maxsplit=-2` and it should be treated the same as `-1`. But you don't need to bother fixing `split()`.

Reply:
I think it's a good idea to fix `split` while I'm in the neighbourhood. :)